drivers/hmis/app/models/hmis/hud/processors/client_processor.rb

```diff
@@ -152,6 +159,19 @@ def self.input_to_multi_fields(input_field, enum_map, not_collected_key, none_fi
       name_attributes.merge(client_attributes)
     end
 
+    private def process_age_range(value)
+      return { dob_data_quality: 99 } if value == "Doesn't know / Prefers not to answer"
```

Comment:
Hrmmm, a predicament. Don't like hard-coding this since it comes from a customizable form, but we don't want to ignore unrecognized input either (since it could lead to silent data loss).

Here is an idea for some logic that might be more future-proof. What do you think? (I'm not necessarily proposing this specific case / syntax, just the logic itself)

```ruby
case value
# value matches a known age range, so process it onto dob with low DQ
when *HudUtility2024.age_range.keys
   # process onto dob
   { dob: approximate_dob(value), dob_data_quality: 2 }
# value matches a known missing data reason, so store that. (not currently expected but future-proofing for desired pick lists)
when *HudUtility2024.dob_data_quality_options.values_at(8,9,99)
   { dob_data_quality:  HudUtility2024.dob_data_quality(value, true) }
# value string-matches a missing data reason (PIT form is an example), so don't raise and store 99
when /doesn't know|prefers not to answer|not collected/i # i'm sure this regex isnt right
   { dob_data_quality: 99 }
# this might be an age range that we don't support processing, so we raise
else
  raise "unexpected value"
end
```

Reply:
The regex you put looks fine actually :)